src/main/java/duke/command/AddTaskCommand.java

```diff
@@ -0,0 +1,42 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.exception.DukeException;
+import duke.task.Task;
+import duke.Ui;
+
+import java.util.Objects;
+
+public class AddTaskCommand implements Command {
+    private Task task;
+
+    public AddTaskCommand(Task task) {
+        this.task = task;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage store) throws DukeException {
+        tasks.addTask(task);
+        ui.print("Got it. I've added this task:\n  " + task.toString() + "\nNow you have " + tasks.size()
+                + " tasks in the list.");
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        AddTaskCommand that = (AddTaskCommand) o;
```

Comment:
Consider using a clearer variable name instead of "that".

Reply:
^